service/handlers_app_user.go

```diff
@@ -0,0 +1,166 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package service
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"time"
+
+	"fmt"
+
+	"github.com/snapcore/snapd/asserts"
+)
+
+var userIndexTemplate = "/static/app_user.html"
+
+// SystemUserRequest is the JSON version of the request to create a system-user assertion
+type SystemUserRequest struct {
+	Email    string `json:"email"`
+	Name     string `json:"name"`
+	Username string `json:"username"`
+	Password string `json:"password"`
+	ModelID  int    `json:"model"`
+	Since    string `json:"since"`
+}
+
+// SystemUserResponse is the response from a system-user creation
+type SystemUserResponse struct {
+	Success      bool   `json:"success"`
+	ErrorCode    string `json:"error_code"`
+	ErrorSubcode string `json:"error_subcode"`
+	ErrorMessage string `json:"message"`
+	Assertion    string `json:"assertion"`
+}
+
+// UserIndexHandler is the front page of the web application
+func UserIndexHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
The main difference is the name of the template, but that could be refactored to be a parameter.